app/code/Magebit/Learning/Block/Product/View/AddToCart.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @copyright Copyright (c) 2024 Magebit (https://magebit.com)
+ * @author    Magebit
+ * @license   GNU General Public License ("GPL") v3.0
+ */
+
+namespace Magebit\Learning\Block\Product\View;
+
+use Magento\Catalog\Model\Product;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+use Magento\Framework\Registry;
+
+class AddToCart extends Template
```

Comment:
Better extend some block class from catalog 
something like vendor/magento/module-catalog/Block/Product/View.php
I can see that there are already methods for getting the product and url